runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -161,292 +175,179 @@ public void changedUpdate(DocumentEvent e)
+		title.setForeground(Color.WHITE);
 
-		return toggleButton;
-	}
+		topPanel.add(title, BorderLayout.NORTH);
+		topPanel.add(searchBar, BorderLayout.CENTER);
 
-	private void highlightButton(JLabel button, boolean enabled)
-	{
-		button.setIcon(enabled ? ON_SWITCHER : OFF_SWITCHER);
-		button.setToolTipText(enabled ? "Disable plugin" : "Enable plugin");
+		onSearchBarChanged();
+		searchBar.requestFocusInWindow();
+		validate();
+		scrollPane.getVerticalScrollBar().setValue(scrollBarPosition);
 	}
 
 	private void onSearchBarChanged()
 	{
 		final String text = searchBar.getText();
 
-		children.values().forEach(this::remove);
+		pluginList.forEach(mainPanel::remove);
 
-		if (text.isEmpty())
-		{
-			children.values().forEach(this::add);
-			revalidate();
-			return;
-		}
+		showMatchingPlugins(pluginList.stream().filter(PluginListItem::isPinned), text);
+		showMatchingPlugins(pluginList.stream().filter(item -> !item.isPinned()), text);
 
-		FuzzySearch.findAndProcess(text, children.keySet(), (k) -> add(children.get(k)));
 		revalidate();
 	}
 
-	@Override
-	public void onActivate()
+	private void showMatchingPlugins(Stream<PluginListItem> listItems, String text)
 	{
-		super.onActivate();
-		if (searchBar.getParent() != null)
+		if (text.isEmpty())
 		{
-			searchBar.requestFocusInWindow();
+			listItems.forEach(mainPanel::add);
+			return;
 		}
-	}
 
-	private void openConfigList()
-	{
-		removeAll();
-
-		JLabel title = new JLabel("Configuration", SwingConstants.LEFT);
-		title.setForeground(Color.WHITE);
-
-		add(title);
-		add(searchBar);
-
-		onSearchBarChanged();
-		searchBar.requestFocusInWindow();
-		JScrollPane scrollbar = getScrollPane();
-		scrollbar.validate();
-		scrollbar.getVerticalScrollBar().setValue(scrollBarPosition);
+		final String[] searchTerms = text.toLowerCase().split(" ");
+		listItems.forEach(listItem ->
+		{
+			if (listItem.matchesSearchTerms(searchTerms))
+			{
+				mainPanel.add(listItem);
+			}
+		});
 	}
 
-	private void changeConfiguration(Config config, JComponent component, ConfigDescriptor cd, ConfigItemDescriptor cid)
+	void openGroupConfigPanel(PluginListItem listItem, Config config, ConfigDescriptor cd)
 	{
-		ConfigItem configItem = cid.getItem();
+		showingPluginList = false;
 
-		if (!Strings.isNullOrEmpty(configItem.warning()))
-		{
-			final int result = JOptionPane.showOptionDialog(component, configItem.warning(),
-				"Are you sure?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE,
-				null, new String[]{"Yes", "No"}, "No");
+		scrollBarPosition = scrollPane.getVerticalScrollBar().getValue();
+		topPanel.removeAll();
+		mainPanel.removeAll();
 
-			if (result != JOptionPane.YES_OPTION)
+		final JLabel topPanelBackButton = new JLabel(BACK_ICON, SwingConstants.LEFT);
+		topPanelBackButton.setPreferredSize(new Dimension(22, 0));
+		topPanelBackButton.addMouseListener(new MouseAdapter()
+		{
+			@Override
+			public void mousePressed(MouseEvent mouseEvent)
```

Comment:
Prefer ActionListener to MouseAdapter as it works with keyboards. If you must use a MouseAdapter use mouseClicked rather than mousePressed

Reply:
You can't add an `ActionListener` to a `JLabel`. I could go with `mouseClicked`, but the clickables on the other panels (most notably `MaterialTab`s) also uses `mousePressed`, and it will result in an inconsistent experience (one triggers on mouse up and one triggers on mouse down).

It also seems to be harder to trigger a `mouseClicked` on a `JLabel` - if your mouse moves between mouse down and mouse up, it doesn't trigger.